spec/classes/virtualbox_spec.rb

```diff
@@ -138,65 +159,8 @@
 
           it { is_expected.to raise_error(Puppet::Error, %r{manage your own repo}) }
         end
-      end
-
-      # Non $::osfamily specific stuff
-      #
-      it { is_expected.to compile.with_all_deps }
-
-      it { is_expected.to contain_package('virtualbox') }
-
-      context 'when managing the kernel' do
-        let(:params) { { 'manage_kernel' => true } }
-
-        it { is_expected.to contain_class('virtualbox::kernel').that_requires('Class[virtualbox::install]') }
-      end
-
-      context 'when not managing the kernel' do
-        let(:params) { { 'manage_kernel' => false } }
-
-        it { is_expected.not_to contain_class('virtualbox::kernel') }
-      end
-
-      context 'when not managing the package' do
-        let(:params) { { 'manage_package' => false } }
-
-        it { is_expected.not_to contain_package('virtualbox') }
-      end
-
-      context 'with a custom package name and version' do
-        let(:params) do
-          {
-            'package_name' => 'virtualbox-custom-package-name',
-            'version' => '4.2'
-          }
-        end
-
-        it { is_expected.to contain_package('virtualbox').with_name('virtualbox-custom-package-name').with_ensure('present') }
-      end
-
-      context 'with the version parameter set to 5.0' do
-        let(:params) { { 'version' => '5.0' } }
-
-        it { is_expected.to contain_class('virtualbox::kernel').with_vboxdrv_command('/usr/lib/virtualbox/vboxdrv.sh') }
-      end
-
-      context 'with the version parameter set to < 5.0' do
-        let(:params) { { 'version' => '4.3' } }
-
-        it { is_expected.to contain_class('virtualbox::kernel').with_vboxdrv_command('/etc/init.d/vboxdrv') }
-      end
-
-      context 'with a custom package name' do
-        let(:params) { { 'package_name' => 'virtualbox-custom-package-name' } }
-
-        it { is_expected.to contain_package('virtualbox').with_name('virtualbox-custom-package-name').with_ensure('present') }
-      end
-
-      context 'with a custom package_ensure value' do
-        let(:params) { { 'package_ensure' => '4.3.16-95972' } }
-
-        it { is_expected.to contain_package('virtualbox').with_ensure('4.3.16-95972') }
+      else
+        it { is_expected.to raise_error(Puppet::Error, %r{not supported}) }
```

Comment:
I think the proper way to do this is:
```suggestion
        it { is_expected.to compile.and_raise_error(%r{not supported}) }
```

Reply:
Ah yes probably. I copy pasted the already used method in the tests.